modules/zip/src/parse-zip/zip64-info-generation.ts

```diff
@@ -0,0 +1,96 @@
+export const signature = new Uint8Array([0x01, 0x00]);
+
+/** info that can be placed into zip64 field */
+type Zip64Options = {
+  /** Original uncompressed file size and Size of compressed data */
+  size?: number;
+  /** Offset of local header record */
+  offset?: number;
+};
+
+/**
+ * creates zip64 extra field
+ * @param options info that can be placed into zip64 field
+ * @returns buffer with field
+ */
+export const createZip64Info = (options: Zip64Options): ArrayBuffer => {
+  const optionsToUse = {
+    ...options,
+    zip64Length: (options.offset ? 1 : 0) * 8 + (options.size ? 1 : 0) * 16
+  };
+
+  let result = new ArrayBuffer(0);
+
+  zip64Fields.forEach((field) => {
+    if (!optionsToUse[field.name ?? ''] && !field.default) {
+      return;
+    }
+    const newValue = new DataView(new ArrayBuffer(field.size));
+    setNumbers[field.size](newValue, 0, optionsToUse[field.name ?? ''] ?? field.default);
+    result = concatArrays(result, newValue.buffer);
+  });
+
+  return result;
+};
+
+/**
+ * concats two ArrayBuffers
+ * @param arr1 first array to concat
+ * @param arr2 second array to concat
+ * @returns concated array
+ */
+export const concatArrays = (arr1: ArrayBuffer, arr2: ArrayBuffer): ArrayBuffer => {
+  const resHeader = new Uint8Array(arr1.byteLength + arr2.byteLength);
+  resHeader.set(new Uint8Array(arr1));
+  resHeader.set(new Uint8Array(arr2), arr1.byteLength);
+  return resHeader.buffer;
+};
+/**
+ * Function to write values into buffer
+ * @param header buffer where to write a value
+ * @param offset offset of the writing start
+ * @param value value to be written
+ */
+type NumberSetter = (header: DataView, offset: number, value: number) => void;
+
+/** functions to write values into buffer according to the bytes amount */
+export const setNumbers: {[key: number]: NumberSetter} = {
```

Comment:
We use verb-noun for function names, so this setNumbers is a little confusing. We also use CAPITAL_CASE for constants. Maybe:
```suggestion
export const NUMBER_SETTERS = {
```